distribution/lib/Standard/Table/0.0.0-dev/package.yaml

```diff
@@ -8,3 +8,5 @@ authors:
 maintainers:
   - name: Enso Team
     email: contact@enso.org
+requires:
+  - Standard.Base
```

Comment:
By specifying `requires` on another package, the package _"sees modules"_ and can `require` them in own `module-info.java` files.

Reply:
How does this relate to the *project-private* and *friend* concept? Table Java code depends on modules from Base Java code, that is fine. But what about Table's Enso code and private stuff in Base? As far as I understand, all the classes inside `polyglot` directory of a project are considered project-private.

My question is: is this `requires` directive the same as *be friends with* directive? If so, should we rename to make it more clear?

Reply:
The biggest problem right now are the multiple `ModuleLayer`s. I am starting to get a feeling that while different modules may use each others modules, it is not possible to use `ServiceLoader` to load services across different _layers_. I still need to verify it, but it seems very likely it is the case.

Reply:
I tried to do some research and write a sample project. Looks like I am not able to register `java.base` into my own layer - other than `ModuleLayer.boot()`. My code:
```java
package org.enso.test.layer.use;

import java.lang.module.Configuration;
import java.lang.module.ModuleFinder;
import java.util.Collections;
import java.util.List;

public class Use {

    public static void main(String[] args) {
        ModuleFinder finder = ModuleFinder.ofSystem();
        // final List<String> moduleNames = finder.findAll().stream().map(x -> x.descriptor().name()).collect(Collectors.toList());
        final List<String> moduleNames = Collections.singletonList("java.base");
        ModuleLayer emptyL = ModuleLayer.empty();
        ClassLoader emptyLoader = null; // ClassLoader.getSystemClassLoader();
        final List<Configuration> pConfs = Collections.singletonList(emptyL.configuration());
        Configuration pConf = Configuration.resolve(finder, pConfs, finder, moduleNames);
        List<ModuleLayer> pLayers = Collections.singletonList(emptyL);
        ModuleLayer.Controller cntrl = ModuleLayer.defineModulesWithOneLoader(pConf, pLayers, emptyLoader);
        System.out.println("Hello World!" + cntrl.layer().modules());
    }
}
```
opinions, @Akirathan?

Reply:
Look for an inspiration in [BenchProcessor.createTruffleModuleLayer](https://github.com/enso-org/enso/blob/develop/lib/scala/bench-processor/src/main/java/org/enso/benchmarks/processor/BenchProcessor.java#L53). I am creating ModuleLayer, and throwing away all the modules that are already present in boot layer there.